src/strkey.rs

```diff
@@ -8,90 +10,41 @@ pub enum DecodeError {
 
 #[derive(Clone, Hash, PartialEq, Eq, Debug)]
 pub enum Strkey {
-    PublicKey(PublicKey),
+    PublicKeyEd25519([u8; 32]),
```

Comment:
❗ I flattened this structure somewhat a few commits ago, but reverted it. What I discovered was that Rust enums are unions, and so this particular arm of the union is not a type that can ever be held in isolation. What that means is it's impossible to hold a `PublicKeyEd25519` in a variable or pass it around, without doing so as the more general `StrKey`. Where this gets awkward is if you want to write a function that only accepts a `PublicKeyEd25519`, or only accepts `PublicKey`s. So I think this change would make it difficult to use specific Strkey types alone.

I do pefer flattened structures in general, but Rust enums do not lend themselves to it. But that is why in the nested structure I have independent structs at the bottom level: It is entirely optional to use for the developer. They can use the `PublicKeyEd25519` alone because it is a `struct`, but then if they need to pass a strkey without knowing what type it will be, they can get a `Strkey` and can unwrap down to the different types.

Reply:
Interesting. I see why that's a problem. There's an additional level of indirection in your code through the `PublickKey` enum, why is that necessary ?

Reply:
I probably got a bit eager. The idea was I might want to decode any type of `G` key, not just `G ed25519` keys. We only have one alg type, so that extra level of indirection is probably unnecessary. We could also remove `Ed25519` and `Sha256` from the names of things. If we ever support other key types in the future, we could make a breaking change to reintroduce something like this.

If you'd like to make that change, could you break this change in half: one change making the non-functional refactor, and one change adding private keys?